### ISSUE TITLE
Async/Await in CareKit

### ISSUE THREAD

**Issue by cadaniel** [carekit-apple/CareKit#597]
Since my recent project that I've been asking about recently is new, I was planning on targeting the iOS 15 launch, and make my app support that to get to use fancy new things like async/await

However, it doesn't seem like care kit has been updated to support async/await yet. Is this in the works and incoming? 

Thanks!

**Comment by erik-apple** [comment]
It's something that we'd like to add soon!

For now, I'd recommend that you write extensions yourself for the subset of method you'd like `async` versions of. Here's an example.

```swift
// MARK: Async Method for OCKAnyReadOnlyTaskStore

@available(iOS 15.0, *)
public extension OCKAnyReadOnlyTaskStore {

    func fetchAnyTasks(query: OCKTaskQuery) async throws -> [OCKAnyTask] {
        try await withCheckedThrowingContinuation { continuation in
            fetchAnyTasks(query: query, callbackQueue: .main) { result in
                continuation.resume(with: result)
            }
        }
    }
}
```

**Comment by cadaniel** [comment]
thanks for the sample!
